rust/pact_ffi/src/verifier/mod.rs

```diff
@@ -339,3 +343,176 @@ ffi_fn! {
+    pub flags: Vec<Argument>
+}
+
+/// External interface to retrieve the options and arguments available when calling the CLI interface,
+/// returning them as a JSON string.
+///
+/// The purpose is to then be able to use in other languages which wrap the FFI library, to implement
+/// the same CLI functionality automatically without manual maintenance of arguments, help descriptions
+/// etc.
+///
+/// # Example structure
+/// ```json
+/// {
+///   "options": [
+///     {
+///       "long": "scheme",
+///       "help": "Provider URI scheme (defaults to http)",
+///       "possible_values": [
+///         "http",
+///         "https"
+///       ],
+///       "default_value": "http"
+///       "multiple": false,
+///     },
+///     {
+///       "long": "file",
+///       "short": "f",
+///       "help": "Pact file to verify (can be repeated)",
+///       "multiple": true
+///     },
+///     {
+///       "long": "user",
+///       "help": "Username to use when fetching pacts from URLS",
+///       "multiple": false,
+///       "env": "PACT_BROKER_USERNAME"
+///     }
+///   ],
+///   "flags": [
+///     {
+///       "long": "disable-ssl-verification",
+///       "help": "Disables validation of SSL certificates",
+///       "multiple": false
+///     }
+///   ]
+/// }
+/// ```
+///
+/// # Safety
+///
+/// Exported functions are inherently unsafe.
+#[no_mangle]
+pub extern "C" fn pactffi_verifier_cli_args() -> *const c_char {
+    let args: Vec<String> = env::args().collect();
+    let program = args[0].clone();
+    let app = args::setup_app(program, clap::crate_version!());
+
+    // Iterate through the args, extracting info from each to then add to a Vector of args
+    let mut options: Vec<Argument> = Vec::new();
+    let mut flags: Vec<Argument> = Vec::new();
+
+    for opt in app.p.opts.iter() {
+        let arg = parse_argument(opt.s.long, opt.s.short, opt.b.help, opt.v.possible_vals.clone(),opt.v.default_val, opt.b.settings.is_set(ArgSettings::Multiple), opt.v.env.clone());
+        options.push(arg);
+    }
+
+    for opt in app.p.flags.iter() {
+        let arg = parse_argument(opt.s.long, opt.s.short, opt.b.help, None, None,opt.b.settings.is_set(ArgSettings::Multiple), None);
+        flags.push(arg);
+    }
+
+
+    let opts_flags = OptionsFlags { options:options, flags: flags};
+    let json = serde_json::to_string(&opts_flags).unwrap();
+    let c_str = CString::new(json).unwrap();
+    c_str.into_raw() as *const c_char
+}
+
+fn parse_argument(long: Option<&str>, short: Option<char>, help: Option<&str>, possible_values: Option<Vec<&str>>, default_value: Option<&OsStr>, multiple: bool, env: Option<(&OsStr, Option<OsString>)>) -> Argument {
+    let mut arg = Argument { short: None, long: None, help: None, possible_values: None, default_value: None, multiple: Some(false), env: None };
+
+    // Long
+    match long {
+        None => {}
+        Some(_val) => {
+            let c_str = CString::new(_val.to_string()).unwrap();
```

Comment:
No need to use CString here. These strings all get put into the final JSON that is sent. All is needed is `arg.long = Some(_val.to_string());`

Variables with underscores are for unused variables where we don't want a compiler warning, so just rename it to `val`.

Reply:
Fixed (that one and the other instances), I think I had gone into Java habits of excessive conversion/casting and failed to tidy after